frontend/src/components/Calendar.vue

```diff
@@ -109,8 +113,8 @@ function handleDateClick(arg) {
   showTransactionDialog.value = true
 }
 
-function handleEventClick(info) {
-  const transaction = transactionsStore.getTransactionById(parseInt(info.event.id))
+async function handleEventClick(info) {
+  const transaction = await transactionsStore.fetchTransactionById(parseInt(info.event.id))
```

Comment:
instead of geting (from the already loaded transactions array), we fetch it from the server to make sure the information is fresh